features/steps/component.go

```diff
@@ -57,35 +82,62 @@ func NewSearchReindexTrackerComponent() (*SearchReindexTrackerComponent, error)
 	searchDataImportedConsumer.StopFunc = func() error { return nil }
 	c.searchDataImportedConsumer = searchDataImportedConsumer
 
-	cfg, err := config.Get()
-	if err != nil {
-		return nil, fmt.Errorf("failed to get config - err: %v", err)
-	}
-	c.cfg = cfg
-
 	initMock := &mock.InitialiserMock{
 		DoGetKafkaConsumersFunc: c.DoGetConsumers,
 		DoGetHealthCheckFunc:    c.DoGetHealthCheck,
+		DoGetHealthClientFunc:   c.DoGetHealthClient,
 		DoGetHTTPServerFunc:     c.DoGetHTTPServer,
 	}
 
 	c.serviceList = service.NewServiceList(initMock)
 
+	// run application in separate goroutine
+	c.startTime = time.Now()
+	c.svc, err = service.Run(c.ctx, c.serviceList, "", "", "", c.errorChan)
+	if err != nil {
+		return nil, fmt.Errorf("failed to run component service - err: %v", err)
+	}
+
 	return c, nil
 }
 
 func (c *SearchReindexTrackerComponent) Close() {
+	// kill application
+	signals := registerInterrupt()
+	signals <- os.Interrupt
 }
 
 func (c *SearchReindexTrackerComponent) Reset() {
 }
 
+// InitAPIFeature initialises the ApiFeature that's contained within a specific JobsFeature.
+func (c *SearchReindexTrackerComponent) InitAPIFeature() *componentTest.APIFeature {
+	c.apiFeature = componentTest.NewAPIFeature(c.InitialiseService)
+
+	return c.apiFeature
+}
+
+// InitialiseService returns the http.Handler that's contained within the component.
+func (c *SearchReindexTrackerComponent) InitialiseService() (http.Handler, error) {
+	return c.HTTPServer.Handler, nil
+}
+
 func (c *SearchReindexTrackerComponent) DoGetHealthCheck(cfg *config.Config, buildTime string, gitCommit string, version string) (service.HealthChecker, error) {
-	return &mock.HealthCheckerMock{
-		AddCheckFunc: func(name string, checker healthcheck.Checker) error { return nil },
-		StartFunc:    func(ctx context.Context) {},
-		StopFunc:     func() {},
-	}, nil
+	componentBuildTime := strconv.Itoa(int(time.Now().Unix()))
+	versionInfo, err := healthcheck.NewVersionInfo(componentBuildTime, gitCommitHash, appVersion)
+	if err != nil {
+		return nil, err
+	}
+	hc := healthcheck.New(versionInfo, cfg.HealthCheckCriticalTimeout, cfg.HealthCheckInterval)
+	return &hc, nil
+}
+
+func (c *SearchReindexTrackerComponent) DoGetHealthClient(name string, url string) *health.Client {
```

Comment:
(suggestion): I guess  its probably a good idea to check the name and url for empty string and return a nil client if it is.

Reply:
Done